test/models/parcel_item_test.rb

```diff
@@ -54,5 +54,20 @@
 
 class ParcelItemTest < ActiveSupport::TestCase
   test_model_actions
-  # Add tests here...
+
+  test "give_incoming doesn't create ProductMovement and ProductLocalization if an error occurs when creating ProductEnjoyment"  do
```

Comment:
Could you rename this test ? You're testing a broader behaviour: that give doesn't create the dependent records if there's an exception during the process. Not specifically ProductMovement/Localization (even if that's what you used to test said behaviour).